packages/new-deploy/scripts/verify.ts

```diff
@@ -0,0 +1,245 @@
+// --chain-id
+// --compiler-version
+// --num-of-optimizations
+
+// address
+// path/to/Contracts.sol:ContractName
+// etherscan-key
+
+import { writeFile, readFile } from 'fs';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import { Verification } from '../src';
+import * as dotenv from 'dotenv';
+import { NomadContext } from '@nomad-xyz/sdk';
+dotenv.config();
+
+type VerificationOutput = Verification & {
+  GUID?: string;
+  verifyCommand?: string;
+  verified?: boolean;
+};
+
+const execAsync = promisify(exec);
+const readFileAsync = promisify(readFile);
+const writeFileAsync = promisify(writeFile);
+
+/// async sleep function
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const etherscanKey = process.env.ETHERSCAN_KEY;
```

Comment:
can you add an error which throws informatively if there's no API key present

Reply:
next PR.